src/parsers/babel_parser.js

```diff
@@ -10,28 +10,27 @@
  */
 
 import {readFileSync} from 'fs';
-
-import {File as BabylonFile, Node as BabylonNode, parse as babylonParse} from 'babylon';
+import {File as BabelFile, Node as BabelNode} from '@babel/types';
+import * as parser from '@babel/parser';
 import type {ParsedNodeType} from './parser_nodes';
 import {NamedBlock, ParsedRange, ParseResult, ParsedNode} from './parser_nodes';
 
-// eslint-disable no-underscore-dangle
-const _getASTfor = (file: string, data: ?string): [BabylonFile, string] => {
+const _getASTfor = (file: string, data: ?string, options: parser.ParserOptions): [BabelFile, string] => {
   const _data = data || readFileSync(file).toString();
-  const config = {plugins: ['*'], sourceType: 'module'};
-  return [babylonParse(_data, config), _data];
+  const config = {...options, sourceType: 'module'};
+  return [parser.parse(_data, config), _data];
 };
 
-export const getASTfor = (file: string, data: ?string): BabylonFile => {
+export const getASTfor = (file: string, data: ?string): BabelFile => {
   const [bFile] = _getASTfor(file, data);
   return bFile;
 };
 
-export const parse = (file: string, data: ?string): ParseResult => {
+export const parse = (file: string, data: ?string, options: parser.ParserOptions): ParseResult => {
```

Comment:
...we could make `options` optional here, too. That way the signature of `parse` would be backward compatible.

Reply:
absolutely, after an optional param cannot be a required param.